### ISSUE TITLE
application failed to started with http-outbound related error

### ISSUE THREAD

**Issue by xiaoshimimi** [RuntimeTools/appmetrics#398]
Here is our test application. 
[joyce-pure.zip](https://github.com/RuntimeTools/appmetrics/files/799574/joyce-pure.zip)

Once we start the application, we got the following error message, and the application stopped: 

```
[root@nc9098038144 joyce-pure]# node app
[Fri Feb 24 08:33:41 2017] com.ibm.diagnostics.healthcenter.loader INFO: Node Application Metrics 2.0.1.201702100931 (Agent Core 3.1.0)
[Fri Feb 24 08:33:41 2017] com.ibm.diagnostics.healthcenter.mqtt INFO: Connecting to broker localhost:1883
[2017-02-24 08:33:41.466] [INFO] knj_log - Configuration updated: deepdive is disabled
[2017-02-24 08:33:41.476] [INFO] knj_log - JsonSender: VCAP_SERVICES env variable not found
[2017-02-24 08:33:41.532] [INFO] knj_log - Configuration updated: deepdive is enabled
/root/Public/joyce-pure/node_modules/appmetrics/probes/http-outbound-probe.js:164
    probeData.req.stop({url: url, statusCode: res.statusCode, contentType:res.headers?res.headers['content-type']:"undefined"});
                 ^

TypeError: Cannot read property 'stop' of undefined
    at HttpOutboundProbe.requestEnd (/root/Public/joyce-pure/node_modules/appmetrics/probes/http-outbound-probe.js:164:18)
    at /root/Public/joyce-pure/node_modules/appmetrics/probes/http-outbound-probe.js:66:22
    at ClientRequest.args.(anonymous function) (/root/Public/joyce-pure/node_modules/appmetrics/lib/aspect.js:26:4)
    at ClientRequest.g (events.js:291:16)
    at emitOne (events.js:96:13)
    at ClientRequest.emit (events.js:188:7)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:474:21)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:99:23)
    at TLSSocket.socketOnData (_http_client.js:363:20)
    at emitOne (events.js:96:13)

```

This seems http-outbound request related.

**Comment by xiaoshimimi** [comment]
we are using appmetrics 2.0.1, there is no error if we use appmetrics 1.0.11

**Comment by sjanuary** [comment]
Seems like requests are being enabled after an http request has started but before it returns, so requestStart is not called and then requestEnd is called with an undefined probeData.  Can be fixed with a null check in requestEnd, will need to implement for all probes.

**Comment by sjanuary** [comment]
http-outbound probe was new in 2.0.1, so that's why this was not seen in 1.0.11

**Comment by xiaoshimimi** [comment]
Is this fix possible to be released in a 2.0.x minus version?

**Comment by tobespc** [comment]
this will get rolled up in the 3.0 release which will go out in a few weeks
